_includes/manuals/2.0/5.1.12_dynflow_scaling.md

```diff
@@ -0,0 +1,172 @@
+Starting with Foreman 2.0 Foreman uses Dynflow backed by Sidekiq by default on all supported platforms. This change is described in greater detail in [Upcoming changes to Dynflow](https://community.theforeman.org/t/upcoming-changes-to-dynflow/14926). The original Dynflow executor service (called `dynflowd`) was replaced by `redis` and a set of `dynflow-sidekiq@*` services. This new deployment model should address issues we have seen when restarting the Dynflow executor. It also allows us scale more easily, dedicate workers to specific queues and increase the overall throughput. The following examples show how to scale Dynflow up in the new model.
+
+### Clean state
+
+Out of the box, foreman ships with orchestrator and a single worker
+If you have Katello, you will get an additional worker for processing of the host queue
+```
+root@modest-gator:~# ls -l /etc/foreman/dynflow/
+total 1
+-rw-r--r--. 1 root foreman 51 May 14 07:35 orchestrator.yml
+-rw-r--r--. 1 root foreman 59 May 14 07:35 worker.yml
+```
+
+The orchestrator consumes items only from the `dynflow_orchestrator` queue and has only one thread. This is by design. *Do not change this*
+```
+root@modest-gator:~# cat /etc/foreman/dynflow/orchestrator.yml
+:concurrency: 1
+:queues:
+  - dynflow_orchestrator
+```
+
+The regular worker has 5 threads and consumes items from
+default and remote_execution queues. It is not recommended
+to change the configuration for this process either, because the changes
+would be overwritten by the installer, if run again.
+
+```
+root@modest-gator:~# cat /etc/foreman/dynflow/worker.yml
+:concurrency: 5
```

Comment:
Out of curiousity: why 5 and not 35? If we know it's thread safe, then we should probably put more work per process.

Reply:
IIRC the original all-in-one executor had 5 threads for the default queue, so in here we just copied it over.

Threads in ruby aren't exactly a silver bullet, so more threads doesn't always mean better performance. We need to find the right balance between the number of processes and number of threads per process.

I admit 5 may be too conservative, but I'd like to wait for the performance team to run some tests before we change it.

Reply:
Generally speaking, our actions (or whatever these are called in dynflow) are I/O bound so I would expect better performance from higher concurrency. Performance team should confirm. We should at least describe this down below what are the sane numbers here.